x/ccv/provider/keeper/partial_set_security.go

```diff
@@ -308,11 +308,44 @@ func (k Keeper) CanValidateChain(ctx sdk.Context, chainID string, providerAddr t
 			!k.IsDenylisted(ctx, chainID, providerAddr))
 }
 
+// FulfillsMinStake returns true if the validator `providerAddr` has enough stake to validate chain `chainID`
+// by checking its staked tokens against the minimum stake required to validate the chain.
+func (k Keeper) FulfillsMinStake(ctx sdk.Context, chainID string, providerAddr types.ProviderConsAddress) bool {
+	minStake, found := k.GetMinStake(ctx, chainID)
+	if !found {
+		return true
+	}
+
+	validator, err := k.stakingKeeper.GetValidatorByConsAddr(ctx, providerAddr.Address)
+	if err != nil {
+		return false
+	}
+
+	// validator has enough stake to validate the chain
+	return validator.GetBondedTokens().GTE(math.NewIntFromUint64(minStake))
+}
+
 // ComputeNextValidators computes the validators for the upcoming epoch based on the currently `bondedValidators`.
-func (k Keeper) ComputeNextValidators(ctx sdk.Context, chainID string, bondedValidators []stakingtypes.Validator) []types.ConsumerValidator {
+func (k Keeper) ComputeNextValidators(ctx sdk.Context, chainID string, bondedValidators []stakingtypes.Validator) []types.ConsensusValidator {
+	// sort the bonded validators by number of staked tokens in descending order
+	sort.Slice(bondedValidators, func(i, j int) bool {
+		return bondedValidators[i].GetTokens().GT(bondedValidators[j].GetTokens())
```

Comment:
I guess since those are bonded validators, `GetTokens()` is the same as `GetBondedTokens()`. Still, I would rather use `GetBondedTokens()` the same way you use it above in `FulfillsMinStake`.

Reply:
Changed, thx

Reply:
is there a need to ensure ordering of vals with same amount of tokens?